chat_stream.go

```diff
@@ -10,13 +10,33 @@ type ChatCompletionStreamChoiceDelta struct {
 	Role         string        `json:"role,omitempty"`
 	FunctionCall *FunctionCall `json:"function_call,omitempty"`
 	ToolCalls    []ToolCall    `json:"tool_calls,omitempty"`
+	Refusal      string        `json:"refusal,omitempty"`
+}
+
+type ChatCompletionStreamChoiceLogprobs struct {
+	Content []ChatCompletionTokenLogprob `json:"content,omitempty"`
+	Refusal []ChatCompletionTokenLogprob `json:"refusal,omitempty"`
+}
+
+type ChatCompletionTokenLogprob struct {
+	Token       string                                 `json:"token"`
+	Bytes       []int64                                `json:"bytes,omitempty"`
+	Logprob     float64                                `json:"logprob,omitempty"`
+	TopLogprobs []ChatCompletionTokenLogprobTopLogprob `json:"top_logprobs"`
+}
+
+type ChatCompletionTokenLogprobTopLogprob struct {
+	Token   string  `json:"token"`
+	Bytes   []int64 `json:"bytes"`
+	Logprob float64 `json:"logprob"`
 }
 
 type ChatCompletionStreamChoice struct {
-	Index                int                             `json:"index"`
-	Delta                ChatCompletionStreamChoiceDelta `json:"delta"`
-	FinishReason         FinishReason                    `json:"finish_reason"`
-	ContentFilterResults ContentFilterResults            `json:"content_filter_results,omitempty"`
+	Index                int                                  `json:"index"`
+	Delta                ChatCompletionStreamChoiceDelta      `json:"delta"`
+	Logprobs             []ChatCompletionStreamChoiceLogprobs `json:"logprobs,omitempty"`
```

Comment:
Hey, thank you for the PR!

OpenAI doc mentions that this is an `object or null,` so I'm curious if we should use the slice here.

Also would appreciate an integration test, making sure all these details work as intended is pretty challenging these days!

<img width="998" alt="image" src="https://github.com/user-attachments/assets/81d7baae-32f0-48fe-89af-13073a162b31">